.github/workflows/publish-packages.yml

```diff
@@ -0,0 +1,89 @@
+name: Publish packages
+
+on:
+  push:
+    # possibly could be improved with a proper regex
+    # ex: https://stackoverflow.com/a/58869470
+    tags:
+      - v*.*.*
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Install moreutils
+        run: brew install moreutils
+
+      - name: Install git-toolbelt
+        run:
+          brew install fzf && brew tap nvie/tap && brew install
+          nvie/tap/git-toolbelt
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.base_ref }}
+
+      - name: Set git tag
+        id: git_tag
+        run: echo "value=${GITHUB_REF#refs/*/}" >> $GITHUB_OUTPUT
+
+      - name: Set release version
+        id: version
+        env:
+          GIT_TAG: ${{ steps.git_tag.outputs.value }}
+        run: echo "value=${GIT_TAG#v}" >> $GITHUB_OUTPUT
+
+      - name: Set npm tag
+        id: npm_tag
+        env:
+          GIT_TAG: ${{ steps.git_tag.outputs.value }}
+        run: |
+          get_tag_if_exists () {
+            if grep -q "-" <<< "$1"; then
+                echo "${1##*-}" | sed 's/[0-9]//g'
+            else 
+                echo "latest"
+            fi
+          }
+          echo value=$(get_tag_if_exists "$GIT_TAG") >> $GITHUB_OUTPUT
```

Comment:
Nice! This is the part where `v0.1.2-beta6` will "infer" `beta` and use it as an NPM tag 👌

I happen to be able to read this, but it's quite hidden/obscure, and I think this convention is really important to document somewhere — did you document this process somewhere already?

Reply:
I did not document it but plan on adding it to the `README.MD` or somewhere obvious for us